grcov parsing of gcda files

rhcu opened the pull request:
Issue #2 
I have made parsing of .gcda files stored in the temporary GCOV directory to store coverage information in `coveralls.json` file with grcov. Used subprocesses to call the shell command in the script and to output the result to `coveralls.json`. Shutil module was used to zip .gcda files.
I am not sure how to check whether the generated report (coveralls.json) is right. Can you suggest a way? I have added it to the commit too.

marco-c commented:
The quickest way is to parse the JSON file with Python and check that some function that is expected to be covered is actually covered.
For example, you can check if https://dxr.mozilla.org/mozilla-central/source/toolkit/xre/nsAppRunner.cpp#3125 is covered (this function is executed at startup in Firefox, so it has to be covered).

marco-c commented on crawler.py:
```diff
@@ -191,10 +193,25 @@ def run_all(driver):
+    shutil.make_archive('code-coverage-gcda', 'zip', gcov_dir)
+    grcov_command = [
+        os.path.join('tools', 'grcov'),
+        '-t', 'coveralls',
```
Let's use `coveralls+`, whose output also contains functions.

marco-c commented on crawler.py:
```diff
@@ -191,10 +193,25 @@ def run_all(driver):
+        '--commit-sha', 'UNUSED'
+    ]
+    with open('coveralls.json', "w+") as outfile:
+        subprocess.call(grcov_command, stdout=outfile)
```
Let's use `check_call` instead of `call`, so the function will raise an exception if grcov fails.

marco-c commented:
Once you've verified that the coveralls.json file is looking good, please remove it from the PR.

marco-c commented on crawler.py:
```diff
@@ -191,10 +193,25 @@ def run_all(driver):
+        '--token', 'UNUSED',
+        '--commit-sha', 'UNUSED'
+    ]
+    with open('coveralls.json', "w+") as outfile:
```
Let's call this `output.json`

rhcu commented:
Ok. Will verify the file with Python script. Thanks for comments. :)

rhcu commented:
Now, there are also functions in the JSON file and deleted this file from the PR.  Also, added a line to delete zipped gcda files from the directory after parsing.
